specification.md

```diff
@@ -319,17 +319,28 @@ Note that different data filters play a big role as to which parts of the states
 evaluated. Reference the 
 [State Data Filtering](#State-Data-Filtering) section for more information about state data filters.
 
-All expressions must follow the [jq](https://stedolan.github.io/jq/) syntax. Jq is a very powerful JSON processor. You 
-can find more information in the [jq manual](https://stedolan.github.io/jq/manual/).
+By default, all workflow expressions should be defined using the [jq](https://stedolan.github.io/jq/) syntax.
+You can find more information on jq in its [manual](https://stedolan.github.io/jq/manual/).
 
-Expressions have the following format:
+Serverless Workflow does not mandate the use of jq and it's possible to use an expression language 
+of your choice with the restriction that a single one must be used for all expressions
+in a workflow definition. If a different expression language needs to be used, make sure to set the workflow 
+`expressionLang` property to identify it to runtime implementations.
+
+Note that using a non-default expression language could lower the portability of your workflow definitions 
+across multiple container/cloud platforms.
+
+All workflow expressions in this document, [specification examples](examples/README.md) as well as [comparisons examples](comparisons/README.md) 
+are written using the default jq syntax.
+
+Workflow expressions have the following format:
 
 ```text
 ${ expression }
 ```
 
-Where `expression` can be either an in-line expression using the [jq](https://stedolan.github.io/jq/) syntax,
-or a reference to a defined [expression function definition](#Using-Functions-For-Expression-Evaluation).
+Where `expression` can be either an in-line expression, or a reference to a 
+defined [expression function definition](#Using-Functions-For-Expression-Evaluation).
```

Comment:
Need to add how to escape in case `"${ notanexpression }"` is supposed to not be an expression, but a string. Suggesting to escape the dollar sign `\$`.

Reply:
An alternative could be to use an object instead of a string, i.e. leave string as string and have a type `StringOrExpression` for e.g. operation parameters, that allows either a string or an object with the parameter `expression` and an optional language:
```
parameters:
  firstParam: "StaticText"
  secondParam:
    expression: "${ .message }"
  thirdParam:
    language: "JSONPath"
    expression: "{{ .message }}"
```
or in designing the object, we could allow the language to be the parameter name:
```
parameters:
  firstParam: "StaticText"
  secondParam:
    jq: "${ .message }"
  thirdParam:
    jsonpath: "{{ .message }}"
```
I'd like the last one most. But feel free to drop it now and we can add this in a follow-up PR

Reply:
in order to model the second scenario we would have to have a defined enum of "possible" expression languages in our schema. don't think we really wanna go that direction as we will never be able to keep up with it and support all adopters

i would like to see the case where atm defining more than one expression language would be needed. i think best maybe is for now to have the top-level expressionLang and to keep enforcing a single expression lang throughout the workflow definition until we have someone bring up a good use case where multiple ones would be needed. wdyt